backends/tofino/bf-p4c/mau/adjust_byte_count.cpp

```diff
@@ -75,7 +75,7 @@ bool AdjustByteCountSetup::Scan::preorder(const IR::MAU::Primitive *prim) {
     const IR::MAU::AttachedMemory *obj = nullptr;
     auto dot = prim->name.find('.');
     cstring method = dot ? cstring(dot + 1) : prim->name;
-    while (dot && dot > prim->name && std::isdigit(dot[-1])) --dot;
+    while (dot && dot > prim->name.c_str() && std::isdigit(dot[-1])) --dot;
```

Comment:
Any reason for this? We should have operator> defined for cstrings right?

Reply:
Yes. But it is a member operator, so `lhs` should be `cstring`.

Reply:
Yeah looking at this I guess the intention was to iterate over the char pointers but I am not 100% sure.

Reply:
It just iterates over cstring backwards skipping digits before dot.